pkg/resources/block_storage/resource_volume.go

```diff
@@ -320,6 +320,12 @@ func (r *ResourceVolume) Read(ctx context.Context, req resource.ReadRequest, res
 	}
 
 	// Read remote state.
+	// Check if ID is empty (resource doesn't exist yet or was just created)
+	if state.ID.ValueString() == "" || state.ID.IsNull() || state.ID.IsUnknown() {
```

Comment:
It cannot be null or unknown in `Read`:
https://developer.hashicorp.com/terraform/plugin/framework/handling-data/accessing-values#when-can-a-value-be-unknown-or-null